imap/message.c

```diff
@@ -522,36 +523,38 @@ EXPORTED int message_parse_mapped(const char *msg_base, unsigned long msg_len,
                                   struct body *body, const char *efname)
 {
     struct msg msg;
+    int r = 0;
 
     msg.base = msg_base;
     msg.len = msg_len;
     msg.offset = 0;
     msg.encode = 0;
 
-    message_parse_body(&msg, body, DEFAULT_CONTENT_TYPE, NULL, efname);
+    r = message_parse_body(&msg, body, DEFAULT_CONTENT_TYPE, NULL, efname);
+    if (r) goto done;
 
     body->filesize = msg_len;
 
     message_guid_generate(&body->guid, msg_base, msg_len);
 
     if (body->filesize != body->header_size + body->content_size) {
         if (efname)
-            /* XXX IOERROR but only LOG_NOTICE?? */
-            xsyslog(LOG_NOTICE, "IOERROR: size mismatch on parse",
-                                "guid=<%s> filename=<%s> "
-                                "filesize=<%" PRIu32 "> bodysize=<%" PRIu32 ">",
-                                message_guid_encode(&body->guid), efname,
-                                body->filesize,
-                                body->header_size + body->content_size);
+            xsyslog(LOG_ERR, "IOERROR: size mismatch on parse",
+                             "guid=<%s> filename=<%s> "
+                             "filesize=<%" PRIu32 "> bodysize=<%" PRIu32 ">",
```

Comment:
I think I probably formatted this xsyslog call originally, but my thinking has changed since.  I'd like to see this:

```suggestion
            xsyslog(LOG_ERR, "IOERROR: size mismatch on parse",
                             "guid=<%s> filename=<%s>"
                             " filesize=<%" PRIu32 "> bodysize=<%" PRIu32 ">",
```

That is, with the separating space at the start of the next line, instead of the end of the previous line.  It's visually a little jarring, which I think draws attention to the fact that the format argument is a single string literal being concatenated together from adjacent pieces (i.e. no comma in between) which are wrapped over multiple lines.

Reply:
OK. I adapted the line for both xsyslog calls.